sklearn/model_selection/_validation.py

```diff
@@ -474,6 +476,13 @@ def _fit_and_predict(estimator, X, y, train, test, verbose, fit_params,
         estimator.fit(X_train, y_train, **fit_params)
     func = getattr(estimator, method)
     predictions = func(X_test)
+    if method in ['predict_proba', 'predict_log_proba']:
+        true_classes = np.unique(y)
+        train_classes = np.unique(y_train)
+        predictions_ = np.zeros((X_test.shape[0], true_classes.shape[0]))
+        for i, j in enumerate(train_classes):
```

Comment:
Once you've encoded the classes outside of this function, I think this loop can be performed as `predictions_[:, estimator.classes_] =  predictions` and in the case of decision_function with binary problems, `predictions_[:, estimator.classes_[-1]] =  predictions`.